### ISSUE TITLE
The OBB Detection， mAP50-95 value is lower than expected

### ISSUE THREAD

**Issue by guoxxxxxxx** [ultralytics/ultralytics#7506]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

When I use yoloobb to train on HRSC2016 dataset. During the training process, mAP50-95 was lower than expected. I checked the output file and found that when predicting, the label drawing was offset from the real object, while the drawing on the training set was correct. At first I thought it was a problem with my data table annotation, so I swapped my training set and validation set. It was found that the Val_label image drawn was still offset, but at the beginning, the train_batch image drawn was correct. 
For the same pictures and labels, the prediction boxes drawn in the verification phase and the training phase are different.
![issue](https://github.com/ultralytics/ultralytics/assets/101721573/0ca303af-3b85-47d0-b336-a33575626f59)

**Comment by frytoli** [comment]
I've come across the same problem: plotted example training images show correct labels, but example validation images show incorrect labels. My validation images kind of look like the label obbs aren't further scaled after the image is batch-resized/formatted (Letterbox-ed). I'm unsure if this is truly affecting training, or if there's potentially a bug in the plotting/viz code.

**Comment by guoxxxxxxx** [comment]
This should not affect the training, but it will affect the evaluation of the model.

**Comment by guoxxxxxxx** [comment]
> I've experienced this same problem: plotted example training images show correct labels, but example validation images show incorrect labels. My validation images kind of look like the label obbs aren't further scaled after the image is batch-resized/formatted (Letterbox-ed). I'm unsure if this is truly affecting training, or if there's potentially a bug in the plotting/viz code.

This should not affect the training, but it will affect the evaluation of the model.

**Comment by glenn-jocher** [comment]
@guoxxxxxxx it sounds like there may be an issue with the visualization process during validation. Rest assured, this should not impact the actual training of your model. However, it could affect the evaluation if the labels are not being processed correctly during the validation phase. We'll look into this to ensure that the visualization accurately reflects the model's performance. Thank you for bringing this to our attention! 🛠️

**Comment by Laughing-q** [comment]
@guoxxxxxxx @frytoli what's the package version you are using? recently we fixed a bug about low val mAP, not sure if it's related.

**Comment by frytoli** [comment]
I was following that thread -- #7401 right? And, I'm cloning the master branch of this repo.

Thank you all for taking a look at this!

**Comment by Laughing-q** [comment]
@frytoli yes please feedback to me if you still encounter the low val map issue. :)

**Comment by frytoli** [comment]
@Laughing-q, my apologies - I think my last comment was a bit confusing. I am unfortunately still encountering this particular issue with the code in your main branch, and I don't know if this is related to #7401. Would you mind helping us out? Thanks!

**Comment by Laughing-q** [comment]
@frytoli I tested with our internal dota8.yaml dataset but the val map and the visual labels on `val_batch0_labels.jpg` looks fine, could you share a minimal dataset so I'm able to reproduce the issue? Thanks!
![val_batch0_labels](https://github.com/ultralytics/ultralytics/assets/61612323/68f5d59a-638a-4d13-9a87-0129ce7a98b8)
![pic-240112-1434-22](https://github.com/ultralytics/ultralytics/assets/61612323/55c90abd-050f-4644-91ba-5b3e27e20045)

**Comment by guoxxxxxxx** [comment]
> @frytoli I tested with our internal dota8.yaml dataset but the val map and the visual labels on `val_batch0_labels.jpg` looks fine, could you share a minimal dataset so I'm able to reproduce the issue? Thanks!我使用我们内部的 dota8.yaml 数据集进行了测试，但 val map 和视觉标签 `val_batch0_labels.jpg` 看起来不错，您能否共享一个最小的数据集，以便我能够重现该问题？谢谢！ ![val_batch0_labels](https://private-user-images.githubusercontent.com/61612323/296148879-68f5d59a-638a-4d13-9a87-0129ce7a98b8.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUwNDE2NTksIm5iZiI6MTcwNTA0MTM1OSwicGF0aCI6Ii82MTYxMjMyMy8yOTYxNDg4NzktNjhmNWQ1OWEtNjM4YS00ZDEzLTlhODctMDEyOWNlN2E5OGI4LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEyVDA2MzU1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgyYzkyZTM1MmFkNjlkNTBiZGIzYzVkZDgzZGYyOTNiMTZhZGE0ZWNlZmRjYTY1OWZiYzhjNDlhZDEzYjI1NmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZSCbyO34IJ3mGKFVkx_kLErjIh8pP23GHgbQR8Wk_T8) ![pic-240112-1434-22](https://private-user-images.githubusercontent.com/61612323/296149087-55c90abd-050f-4644-91ba-5b3e27e20045.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDUwNDE2NTksIm5iZiI6MTcwNTA0MTM1OSwicGF0aCI6Ii82MTYxMjMyMy8yOTYxNDkwODctNTVjOTBhYmQtMDUwZi00NjQ0LTkxYmEtNWIzZTI3ZTIwMDQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTEyVDA2MzU1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1MDMwYjI3NDQ5NjVlYTU0NTQyNjA1MGNjNzI4ZGNiZmE4YzEwYjc1ZDhmNTJhMDQ5OWFjZTVjY2Q5NDc5M2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.GieBDTJ4wQ-4cSHmCJWYxO4W27uqNftQm75FNCnWmjI)

I just tested it on the main branch, and the problem is still not solved. I can share a minimal data set to help you reproduce the problem.

link：https://pan.baidu.com/s/1OpjfvjmVo4gYBOxoH9BxTA?pwd=tekn 
pwd：tekn

**Comment by Laughing-q** [comment]
@guoxxxxxxx Thanks! I'll test it later

**Comment by frytoli** [comment]
Thank you both for your help, and enjoy your break @Laughing-q!
